ocfweb/docs/docs/services/hpc/slurm.md

```diff
@@ -0,0 +1,142 @@
+job's output to `stdout` are put into your home directory.
+
+#### When to use `srun` versus `sbatch`
+
+If your job requires interactivity or inputs from the terminal, or you need a
+terminal to test or experiment, use `srun`. Otherwise, use `sbatch`, as you
+don't have to keep your terminal open until the job runs.
+
+### Resource options
+
+*Some terminology:* Slurm refers to a process as a "task". Even if a single
+process is using multiple threads/CPUs, it still counts as one task.
+
+**By default, without any flags, a job you submit will be allocated one CPU,
+100 MB of RAM, and no GPUs, and will run for at maximum 2 days. In order to
+allocate more resources and time to your job, you must set one or more of these
+flags:**
+
+* `-n` / `--ntasks`:
+    - The number of tasks/processes to allocate. **Default is 1.**
+* `-c` / `--cpus-per-task`:
+    - The number of CPUs to allocate per task. **Default is 1.**
+* `--mem`:
+    - The total amount of RAM to allocate. By default, the number supplied is
+assumed to megabytes. However, the prefixes `K`, `M`, `G`, and `T` can be
+appended to the number instead. For example, to allocate 5 gigabytes of ram,
+use `--mem=5G`. **Default is 100 megabytes.**
+* `--gres` **(Optional)**:
+    - Allocates some GPUs to your job. The format is `--gres=gpu:[optional
+type]:[number to allocate]`. For example, to allocate 2 GPUs of any type, you
+would include `--gres=gpu:2`. To allocate two Nvidia 1080Ti GPUs (our only type
+right now), you would include `--gres=gpu:nv1080:2`. **No default.**
+* `--t` / `--time` **(Optional)**:
+    - The maximum amount of time your job can take before Slurm forcefully
+kills it. Acceptable time formats include "minutes", "minutes:seconds",
+"hours:minutes:seconds", "days-hours", "days-hours:minutes" and
+"days-hours:minutes:seconds". You cannot set the time limit greater than the
+**default, which is 2 days.**
+
+### Using `srun`
+
+On `hpcctl`:
+
+```
+srun [command-line flags] [command to run]
+```
+
+For example, to run a job that uses 4 CPUs, 8 GB of RAM, and 1 GPU:
+
+```
+bzh@hpcctl:~$ srun --ntasks=1 --cpus-per-task=4 --mem=8G --gres=gpu:1 echo
+"Hello world!"
+
+Hello world!
+```
+
+#### Running an interactive terminal
+
+To start up an interactive terminal on a compute node, use the `--pty [your
+terminal of choice]` flag. For most everyone, you'll be using `bash`, so to
+start an interactive terminal on a node, run:
+
+```
+srun [other command-line flags] --pty bash
+```
+
+### Using `sbatch`
+
+A Slurm batch script is functionally the same as a regular `bash` script: The
+`bash` shebang at the start, and script after.
+
+However, to pass options into SLURM, you'll need to add some special comment
+lines, which are in the format `#SBATCH [command-line flag]=[value]`. **They
+must be after the shebang but before any non-comments**.
+
+For example, a batch script which uses 4 CPUs, 8 GB of RAM, and 1 GPU has its
+contents as:
+
+```
+ #!/bin/bash
```

Comment:
Why the leading spaces?

Reply:
It'll trigger the pre-commit hook about "No top-level headers" otherwise.

Reply:
There should be another way to silence that, no?

Reply:
It's a [simple regex](https://github.com/ocf/ocfweb/blob/master/.pre-commit-config.yaml#L58), so I don't think so? Unless pre-commit has a mechanism to silence a check for a single file, but a quick search didn't reveal anything.